end-to-end-untargeted-metabolomics.Rmd

```diff
@@ -122,123 +141,132 @@ The *mzML* files with the raw MS data are located within the *data/mzML* folder
 of this repository. [ideally, they should be added and then downloaded from
 MetaboLight].
 
-Data import from metaboloight first need to be extracted using the *isatabr* 
+Data import from metaboloight first need to be extracted using the *isatabr*
 package. (hjere there I need to write an import command specific fro ISA-tab file.)
 
 ```{r import}
 #' read the sample descriptions from an xlsx sheet
-pd <- read_xlsx("data/phenodata.xlsx") |>
+pd <- read_xlsx("data/phenodata-lc-ms.xlsx") |>
     as.data.frame()
 
 #' Import the data
 #' Massive data bank extraction but for now not.
+#' MZML_PATH <- "data/metabolights/mzML/"
 MZML_PATH <- "C:/Users/plouail/OneDrive - Scientific Network South Tyrol/end-to-end_worflow/data/mzML_files/MS1"
 
-data <- readMsExperiment(paste0(MZML_PATH, "/", pd$mzML_file), sampleData = pd)
+data <- readMsExperiment(file.path(MZML_PATH, "/", pd$mzML_file),
+                         sampleData = pd)
 ```
 
-We next need to set up parallel processing to speed up the analysis. This can be
-done using the *BiocParallel* package, which provides a unified interface for 
-parallel processing in R. Here, we use the *MulticoreParam* and *SnowParam*
-functions to set up parallel processing with 2 cores. The code below checks the
-operating system and registers the parallel processing backend accordingly.
-This is essential for speeding up the analysis, especially when working with 
-large datasets.
+We next configure the parallel processing setup. Most functions from the *xcms*
+package allow per-sample parallel processing, which can improve the performance
+of the analysis, especially for large data sets. *xcms* and all packages from
+the *RforMassSpectrometry* package ecosystem use the parallel processing setup
+configured through the `r Biocpkg("BiocParallel")` Bioconductor package. With
+the code below we use a *fork-based* parallel processing on unix system, and a
+*socket-based* parallel processing on the Windows operating system.
 
 ```{r par-process}
 #' Set up parallel processing using 2 cores
 if (.Platform$OS.type == "unix") {
-    register(bpstart(MulticoreParam(2)))
```

Comment:
removed `bpstart`, because that can sometimes lead to issues (if e.g. Windows decided to stop the socket)